core/chains/evm/orm.go

```diff
@@ -0,0 +1,104 @@
+package evm
+
+import (
+	"github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/sqlx"
+	null "gopkg.in/guregu/null.v4"
+)
+
+type ORM interface {
+	CreateChain(id utils.Big, config types.ChainCfg) (types.Chain, error)
+	DeleteChain(id utils.Big) error
+	Chains(offset, limit int) ([]types.Chain, int, error)
+	CreateNode(data NewNode) (types.Node, error)
+	DeleteNode(id int64) error
+	Nodes(offset, limit int) ([]types.Node, int, error)
+	NodesForChain(chainID utils.Big, offset, limit int) ([]types.Node, int, error)
+}
+
+type orm struct {
+	db *sqlx.DB
+}
+
+var _ ORM = (*orm)(nil)
+
+func NewORM(db *sqlx.DB) ORM {
+	return &orm{db}
+}
+func (o *orm) CreateChain(id utils.Big, config types.ChainCfg) (chain types.Chain, err error) {
+	sql := `INSERT INTO evm_chains (id, cfg, created_at, updated_at) VALUES ($1, $2, now(), now()) RETURNING *`
+	err = o.db.Get(&chain, sql, id, config)
+	return chain, err
+}
+func (o *orm) DeleteChain(id utils.Big) error {
+	sql := `DELETE FROM evm_chains WHERE id = $1`
+	_, err := o.db.Exec(sql, id)
+	// TODO: check result.RowsAffected?
```

Comment:
Probably would be nice to throw an error if you try to delete something that doesn't exist